SoftLayer/CLI/hardware/detail.py

```diff
@@ -107,6 +110,24 @@ def cli(env, identifier, passwords, price):
             pass_table.add_row([item['username'], item['password']])
         table.add_row(['remote users', pass_table])
 
+    if components:
+        components = hardware.get_components(identifier)
+        components_table = formatting.Table(['name', 'version', 'date', 'component-type'])
+        components_table.align['date'] = 'l'
+        ids = []
+        for component in components:
+            if component['id'] not in ids:
```

Comment:
Are there duplicate `ids` in the components list? I would assume this check wouldn't be needed. But if this check is preventing duplicate line entries or something feel free to keep it in there.